### Importar Librerias

In [1]:
import pandas as pd
import numpy as np

### Carga de datasets y Exploracion inicial

In [2]:
df = pd.read_csv('../datasets/NY-House-Dataset.csv')

In [3]:
print("--Exploracion Inicial del Dataset--")
print('Dimensiones del Dataset:', df.shape)
print('\nColumnas del Dataset')
print(list(df.columns))
print('\nTipos de datos de las columnas del dataset')
print(df.dtypes)
print('\nPrimeras filas del dataset')
print(df.head())

--Exploracion Inicial del Dataset--
Dimensiones del Dataset: (4801, 17)

Columnas del Dataset
['BROKERTITLE', 'TYPE', 'PRICE', 'BEDS', 'BATH', 'PROPERTYSQFT', 'ADDRESS', 'STATE', 'MAIN_ADDRESS', 'ADMINISTRATIVE_AREA_LEVEL_2', 'LOCALITY', 'SUBLOCALITY', 'STREET_NAME', 'LONG_NAME', 'FORMATTED_ADDRESS', 'LATITUDE', 'LONGITUDE']

Tipos de datos de las columnas del dataset
BROKERTITLE                        str
TYPE                               str
PRICE                            int64
BEDS                             int64
BATH                           float64
PROPERTYSQFT                   float64
ADDRESS                            str
STATE                              str
MAIN_ADDRESS                       str
ADMINISTRATIVE_AREA_LEVEL_2        str
LOCALITY                           str
SUBLOCALITY                        str
STREET_NAME                        str
LONG_NAME                          str
FORMATTED_ADDRESS                  str
LATITUDE                       float64
LONGI

### Busqueda de valores faltantes y Duplicados

In [4]:
print("Busqueda de valores faltantes")
print(df.isnull().sum())
print('\nBusqueda de datos duplicados', df.duplicated().sum())


Busqueda de valores faltantes
BROKERTITLE                    0
TYPE                           0
PRICE                          0
BEDS                           0
BATH                           0
PROPERTYSQFT                   0
ADDRESS                        0
STATE                          0
MAIN_ADDRESS                   0
ADMINISTRATIVE_AREA_LEVEL_2    0
LOCALITY                       0
SUBLOCALITY                    0
STREET_NAME                    0
LONG_NAME                      0
FORMATTED_ADDRESS              0
LATITUDE                       0
LONGITUDE                      0
dtype: int64

Busqueda de datos duplicados 214


In [17]:
#Duplicados completos
duplicados_completo = df[df.duplicated(keep=False)]

print("Filas Involucradas en duplicados")
print(len(duplicados_completo))
print("\nNumero de duplicados addicionales")
print(df.duplicated().sum())
print("\nDuplicados Encontrados:")
print(duplicados_completo.sort_values('FORMATTED_ADDRESS').head(20).to_string())



Filas Involucradas en duplicados
427

Numero de duplicados addicionales
214

Duplicados Encontrados:
                                  BROKERTITLE                        TYPE    PRICE  BEDS  BATH  PROPERTYSQFT                       ADDRESS                     STATE                                 MAIN_ADDRESS ADMINISTRATIVE_AREA_LEVEL_2         LOCALITY      SUBLOCALITY        STREET_NAME        LONG_NAME                                FORMATTED_ADDRESS   LATITUDE  LONGITUDE
3179        Brokered by Ditommaso Real Estate                     Pending   769000     3   2.0   1170.000000               10 Neal Dow Ave   Staten Island, NY 10314       10 Neal Dow AveStaten Island, NY 10314               United States         New York  Richmond County      Staten Island  Neal Dow Avenue    10 Neal Dow Ave, Staten Island, NY 10314, USA  40.622260 -74.134984
3176        Brokered by Ditommaso Real Estate                     Pending   769000     3   2.0   1170.000000               10 Neal Dow Ave   

In [19]:
#Eliminar los duplicados encontrados
df = df.drop_duplicates().reset_index(drop=True)
print("Duplicados restantes:", df.duplicated().sum())
print("Nuevas dimensiones del dataset:", df.shape)

Duplicados restantes: 0
Nuevas dimensiones del dataset: (4587, 17)


### Análisis y tratamiento de registros 

Durante la exploración inicial se identificaron **0** valores faltantes, pero hubieron **214 registros duplicados** mediante `duplicated().sum()`.

Antes de eliminarlos, se realizó una inspección utilizando `duplicated(keep=False)` para visualizar tanto los registros originales como sus respectivas copias. Se revisaron diferentes muestras de los registros duplicados y se comprobó que las filas contenían los mismos valores en todas las variables, incluyendo precio y características de la propiedad.

Al tratarse de duplicados completos y no de propiedades con características o precios diferentes, se decidió eliminar las copias mediante `drop_duplicates()`.

Finalmente, se restableció el índice del DataFrame con `reset_index(drop=True)` y se verificó nuevamente la cantidad de duplicados para confirmar que el dataset no conservara registros repetidos.

In [35]:
columnas_categoricas = df.select_dtypes(include='str').columns
columnas_numericas = df.select_dtypes(include=['int','float']).columns

print("Columnas Categoricas del dataset:")
print(columnas_categoricas)
print('\nColumnas Numericas del datset:')
print(columnas_numericas)


Columnas Categoricas del dataset:
Index(['BROKERTITLE', 'TYPE', 'ADDRESS', 'STATE', 'MAIN_ADDRESS',
       'ADMINISTRATIVE_AREA_LEVEL_2', 'LOCALITY', 'SUBLOCALITY', 'STREET_NAME',
       'LONG_NAME', 'FORMATTED_ADDRESS'],
      dtype='str')

Columnas Numericas del datset:
Index(['PRICE', 'BEDS', 'BATH', 'PROPERTYSQFT', 'LATITUDE', 'LONGITUDE'], dtype='str')


In [37]:
#Guardar dataset limpio

df.to_csv('../datasets/NY_House_Dataset_clean.csv', index=False)